sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -101,8 +101,17 @@ import org.apache.spark.sql.types.IntegerType
  */
 object RewriteDistinctAggregates extends Rule[LogicalPlan] {
 
+  private def mayNeedtoRewrite(exprs: Seq[Expression]): Boolean = {
+    val distinctAggs = exprs.flatMap { _.collect {
+      case ae: AggregateExpression if ae.isDistinct => ae
+    }}
```

Comment:
are you trying to filter? exprs.filter(_. isDistinct_ does the job!

Reply:
All of them are not `AggregateExpr`, so we need to pick up `AggregateExpr`s from `exprs` first, then apply  cast&filter.